Source/FieldSolver/SpectralSolver/SpectralAlgorithms/PsatdAlgorithm.cpp

```diff
@@ -270,33 +288,46 @@ PsatdAlgorithm::pushSpectralFields (SpectralFieldData& f) const
                 const Complex Jy_new = fields(i,j,k,Idx.Jy_new);
                 const Complex Jz_new = fields(i,j,k,Idx.Jz_new);
 
-                const Complex F_old = fields(i,j,k,Idx.F);
-                const Complex G_old = fields(i,j,k,Idx.G);
+                fields(i,j,k,Idx.Ex) += -X1 * (Jx_new - Jx) / dt;
+
+                fields(i,j,k,Idx.Ey) += -X1 * (Jy_new - Jy) / dt;
+
+                fields(i,j,k,Idx.Ez) += -X1 * (Jz_new - Jz) / dt;
+
+                fields(i,j,k,Idx.Bx) += I * X2/c2 * (ky * (Jz_new - Jz) - kz * (Jy_new - Jy));
+
+                fields(i,j,k,Idx.By) += I * X2/c2 * (kz * (Jx_new - Jx) - kx * (Jz_new - Jz));
+
+                fields(i,j,k,Idx.Bz) += I * X2/c2 * (kx * (Jy_new - Jy) - ky * (Jx_new - Jx));
```

Comment:
Just a bit of taste - at some point too many newlines can hinder readability again :)
```suggestion
                fields(i,j,k,Idx.Ex) += -X1 * (Jx_new - Jx) / dt;
                fields(i,j,k,Idx.Ey) += -X1 * (Jy_new - Jy) / dt;
                fields(i,j,k,Idx.Ez) += -X1 * (Jz_new - Jz) / dt;

                fields(i,j,k,Idx.Bx) += I * X2/c2 * (ky * (Jz_new - Jz) - kz * (Jy_new - Jy));
                fields(i,j,k,Idx.By) += I * X2/c2 * (kz * (Jx_new - Jx) - kx * (Jz_new - Jz));
                fields(i,j,k,Idx.Bz) += I * X2/c2 * (kx * (Jy_new - Jy) - ky * (Jx_new - Jx));
```